prime-router/src/main/kotlin/azure/FHIRFlowFunctions.kt

```diff
@@ -16,6 +26,38 @@ const val fhirQueueName = "fhir-raw-received"
  * Process will work through all the reports waiting in the 'process' queue
  */
 class FHIRFlowFunctions : Logging {
+    /**
+     * An azure function for processing an HL7 message into of FHIR
+     */
+    @FunctionName("convert")
+    @StorageAccount("AzureWebJobsStorage")
+    fun convert(
+        @HttpTrigger(
+            name = "convert",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.FUNCTION,
+            route = "\$convert-data"
+        ) request: HttpRequestMessage<String?>,
+    ): HttpResponseMessage {
+        val responseBuilder = request.createResponseBuilder(HttpStatus.BAD_REQUEST)
+        try {
+            val hl7messages = HL7.deserialize(request.body!!)
+            val body = buildString {
+                hl7messages.forEach { hl7 ->
+                    val fhir = FHIR.translate(hl7)
+                    appendLine(fhir.encode())
+                }
+            }
+            responseBuilder.status(HttpStatus.OK)
+            responseBuilder.header(HttpHeaders.CONTENT_TYPE, "application/fhir+json")
```

Comment:
Maybe use the already available response builders in HTTP utilities.  This is duplicate code.